packages/scope-manager/src/referencer/TypeVisitor.ts

```diff
@@ -197,7 +197,21 @@ class TypeVisitor extends Visitor {
   protected TSMappedType(node: TSESTree.TSMappedType): void {
     // mapped types key can only be referenced within their return value
     this.#referencer.scopeManager.nestMappedTypeScope(node);
-    this.visitChildren(node);
+    this.#referencer
+      .currentScope()
+      .defineIdentifier(
+        node.key,
+        new TypeDefinition(
+          node.key,
+          node.parent as
+            | TSESTree.TSInterfaceDeclaration
+            | TSESTree.TSTypeAliasDeclaration
+            | TSESTree.TSTypeParameter,
```

Comment:
I added #6225 to the v7 milestone. It's never too early to get excited about the next major version! 😄